core/src/main/java/org/wildfly/channel/ManifestRef.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.channel;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+
+public class ManifestRef {
```

Comment:
Maybe rename it to `ManifestCoordinate` and create a superclass common to https://github.com/wildfly-extras/wildfly-channel/blob/main/maven-resolver/src/main/java/org/wildfly/channel/maven/ChannelCoordinate.java

We want to be able to locate a channel or a manifest using either an URL or a GA(V)

Reply:
I added `ChannelMetadataCoordinate` as superclass to both `ChannelCoordinate` and `ChannelManifestCoordinate`